src/GregClient/Utility/FileUtilities.cs

```diff
@@ -79,21 +79,53 @@ public static byte[] GetFileHash(FileInfo fInfo)
             return hashValue;
         }
 
+        /// <summary>
+        /// Gets the full path of the user temporary folder
+        /// </summary>
+        /// <returns>String with the full path of the user temporary folder</returns>
         public static string GetTempFolder()
         {
             return Path.GetDirectoryName(Path.GetTempPath());
         }
 
+        /// <summary>
+        /// Generates a full path under the temporary folder with the prefix 'gregPkg'
+        /// </summary>
+        /// <returns>String with the full path</returns>
         public static string GetTempZipPath()
         {
-            var tempFolder = GetTempFolder();
-            return Path.Combine(tempFolder, "gregPkg" + DateTime.Now.Millisecond.ToString() + ".zip");
+            string tempZipPath = GetZipPath("gregPkg"); 
+
+            return tempZipPath;
         }
 
+        /// <summary>
+        /// Generates a full output path under the temporary folder with the prefix 'gregPkgOutput'
+        /// </summary>
+        /// <returns>String with the full path</returns>
         public static string GetTempZipOutputPath()
+        {
+            string tempZipOutputPath = GetZipPath("gregPkgOutput");
+
+            return tempZipOutputPath;
+        }
+
+        /// <summary>
+        /// Generates a full path under the temporary folder of a zip file with a format like this prefix[GUID].zip
+        /// </summary>
+        /// <param name="prefixFileName">Prefix of the file</param>
+        /// <returns>String with the full path</returns>
+        private static string GetZipPath(string prefixFileName)
         {
             var tempFolder = GetTempFolder();
-            return Path.Combine(tempFolder, "gregPkgOutput" + DateTime.Now.Millisecond.ToString());
+            string zipPath = Path.Combine(tempFolder, String.Format("{0}{1}.zip", prefixFileName, Guid.NewGuid().ToString()));
+
+            while (File.Exists(zipPath))
```

Comment:
@alfredo-pozo this looks solid, but this type of thing always terrifies me :) Can you imagine a situation where this will get stuck?

Once we bring this in we will need to publish a nuget and then update the references in Dynamo core repo - we'll need to check tests there once that is done as there are obviously tests using these methods.
I believe myself or @zeusongit can do the nuget publish if you do not have access.

Reply:
Can we just assume that guid generation will create a unique string and not do the test?  Since we're not creating the folder here, we're still going to have the same collision problem as before if something somehow blows up and guid generation is not unique (but then we'll likely have other problems).

Reply:
@mjkkirschner I'm not sure if there's an scenario where that could happen, since the repetition it is with a very low probability as we are using Guid's now, instead of miliseconds.

```
How unique is a GUID?
128-bits is big enough and the generation algorithm is unique enough that if 1,000,000,000 GUIDs per second were generated for 1 year the probability of a duplicate would be only 50%. Or if every human on Earth generated 600,000,000 GUIDs there would only be a 50% probability of a duplicate.
```

Like @gregmarr said, we can remove the validation if you prefer, I'm ok with whatever you decide? 

About the publishing, as we are doing with another packages, I don't have the access token to publish the package with the  DynamoVisualProgramming user, so I think it's better if you or @zeusongit want to do it

Reply:
okay, sounds good Alfredo, lets get rid of the while loop, then feel free to merge this, and Ashish or I will publish.

Reply:
The change is ready @mjkkirschner , BTW, I don't have permissions to do the merge